qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -411,18 +411,34 @@ class WebEngineHistory(browsertab.AbstractHistory):
     def current_idx(self):
         return self._history.currentItemIndex()
 
-    def back(self):
-        self._history.back()
+    def back(self, count):
+        idx = self.current_idx() - count
+        if idx >= 0:
+            self.goToItem(self.itemAt(idx))
+        else:
+            self.goToItem(self.itemAt(0))
+            raise IndexError
 
-    def forward(self):
-        self._history.forward()
+    def forward(self, count):
+        idx = self.current_idx() + count
+        if idx < len(self):
+            self.goToItem(self.itemAt(idx))
+        else:
+            self.goToItem(self.itemAt(len(self) - 1))
+            raise IndexError
```

Comment:
I'd prefer `raise browsertab.WebTabError("At beginning of history")` or so - then the command-handler can simply re-raise it, and if we expose this to a plugin API in the future, they also get a good error message.